irodsConnector/session.py

```diff
@@ -170,90 +201,30 @@ def password(self) -> str:
             file. Can be a PAM negotiated password.
 
         """
-        if not self._password:
-            irods_auth_file = os.environ.get(
-                'IRODS_AUTHENTICATION_FILE', None)
-            if irods_auth_file is None:
-                irods_auth_file = utils.path.LocalPath(
-                    '~/.irods/.irodsA').expanduser()
-            if irods_auth_file.exists():
-                with open(irods_auth_file, encoding='utf-8') as authfd:
-                    self._password = irods.password_obfuscation.decode(
-                        authfd.read())
-        return self._password
-
-    @password.setter
-    def password(self, password: str):
-        """iRODS password setter method.
-
-        Pararmeters
-        -----------
-        password: str
-            Unencrypted iRODS password.
-
-        """
-        self._password = password
-
-    @password.deleter
-    def password(self):
-        """iRODS password deleter method.
-
-        """
-        self._password = ''
+        irods_auth_file = os.environ.get(
+            'IRODS_AUTHENTICATION_FILE', None)
+        if irods_auth_file is None:
+            irods_auth_file = utils.path.LocalPath('~/.irods/.irodsA').expanduser()
+        if irods_auth_file.exists():
+            with open(irods_auth_file, encoding='utf-8') as authfd:
+                self._password = irods.password_obfuscation.decode(authfd.read())
+                return self._password
 
     @property
-    def session(self) -> irods.session.iRODSSession:
-        """iRODS session creation.
+    def irods_session(self):
+        if self._irods_session:
+            return self._irods_session
+        return None
```

Comment:
This just returns the irods_session, so probably unnecessary after renaming the private attribute to the public attribute.